File: ev-battery-frontend/src/types/vehicle.ts
export interface VehicleFormData {
  nickname: string;
  make: string;
  model: string;
  variant: string;
  vin: string;
  year: number | null;
  batteryCapacityKwh: number | null;
  usableCapacityKwh?: number | null;
  ratedRangeKm?: number | null;
  fastChargeSupported: boolean;
  maxAcPowerKw?: number | null;
  maxDcPowerKw?: number | null;
  chemistry: 'NMC' | 'LFP';
  vehicleType: 'Car' | 'Bike' | 'Scooter';
}

export interface Vehicle extends VehicleFormData {
  id: string;
  userId: string;
  createdAt: string;
}

export interface VehicleDraft {
  id: string;
  status: 'PROCESSING' | 'READY' | 'FAILED';
  extractedData?: Partial<VehicleFormData>;
}

export interface UploadDocumentResponse {
  draftId: string;
}

File: AI-battery/src/main/java/com/ev/AI_battery/model/Vehicle.java
package com.ev.AI_battery.model;

import com.ev.AI_battery.model.VehicleType;
import jakarta.persistence.*;
import lombok.*;

@Entity
@Table(name = "vehicles")
@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
public class Vehicle {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // Ownership
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private VehicleType vehicleType = VehicleType.CAR;

    // Core identifiers
    private String nickname;
    private String make;
    private String model;
    private String variant;
    private Integer year;
    private String vin;

    // Battery details
    private Double batteryCapacityKwh;
    private Double usableCapacityKwh;
    private String chemistry; // NMC / LFP
    private Double ratedRangeKm;

    private Boolean fastChargeSupported;
    private Double maxAcPowerKw;
    private Double maxDcPowerKw;

    private Boolean isDefault = false;
}
